### ISSUE TITLE
Label.SetText() doesn't update widget's text

### ISSUE THREAD

**Issue by frei-0xff** [fyne-io/fyne#3048]
### Describe the bug:
I'm trying to update simple label's text many times relatively fast.
After several updates label's text stops changing.
If I uncomment calls to Hide() and Show() after SetText() everything works as expected and
label's text keeps changind indefinetely.
Tested on Linux only.

### To Reproduce:
Run the sample program and wait for label's text to stop updating.

### Screenshots:
![box-220607-1948-44](https://user-images.githubusercontent.com/2990839/172439759-02d3a9c3-cab9-4960-aaf6-48ad589d6396.gif)

### Example code:

```go
package main

import (
	"fmt"
	"time"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello World")

	l := widget.NewLabel("")
	w.SetContent(l)

	ch := make(chan string)
	go func() {
		for s := range ch {
			l.SetText(s)
			// l.Hide()
			// l.Show()
		}
	}()
	for i := range []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10} {
		go func(i int) {
			for {
				ch <- fmt.Sprintf("Thread %d: %s", i, time.Now().String())
				time.Sleep(100 * time.Millisecond)
			}
		}(i)
	}
	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** 5.15.45
 - **Go version:** 1.18.3
 - **Fyne version:** 2.1.4

**Comment by andydotxyz** [comment]
I think this might be a duplicate of #2867

**Comment by frei-0xff** [comment]
Fixed in v2.2.0
